aws/resource_aws_servicecatalog_product.go

```diff
@@ -0,0 +1,446 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsServiceCatalogProduct() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceCatalogProductCreate,
+		Read:   resourceAwsServiceCatalogProductRead,
+		Update: resourceAwsServiceCatalogProductUpdate,
+		Delete: resourceAwsServiceCatalogProductDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Update: schema.DefaultTimeout(15 * time.Minute),
+			Delete: schema.DefaultTimeout(15 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"distributor": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"product_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"has_default_path": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"product_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
now that enum slices exist lets change this.
ValidateFunc: validation.StringInSlice(servicecatalog.ProductType_Values(), false),